hub-saml/src/main/java/uk/gov/ida/saml/hub/domain/IdpIdaStatus.java

```diff
@@ -79,16 +79,16 @@ public String toString() {
             '}';
     }
 
-    public static class IdpIdaStatusFactory {
+    public static class IdpIdaStatusFactory implements AuthenticationStatusFactory<Status, IdpIdaStatus> {
         public IdpIdaStatus create(
                 final IdpIdaStatus.Status statusCode,
-                final Optional<String> message) {
+                final String message) {
 
             if (!statusCode.equals(Status.RequesterError)) {
                 return new IdpIdaStatus(statusCode);
             }
 
-            return new IdpIdaStatus(statusCode, message);
+            return new IdpIdaStatus(statusCode, Optional.ofNullable(message));
```

Comment:
Really dislike this optional as parameter/field stuff. I know its pre-existing code but eurgh...

Reply:
Yeah Optional shouldn't be used as a param in Java world and it's definitely misused a lot in our code base but again - separate story and issue. The `create` method is now part of an interface and had to be unified, hence the way it's called here has changes, but the logic and behaviour are the same.